modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderLocalServiceImpl.java

```diff
@@ -358,8 +358,9 @@ public CommerceOrder addOrUpdateCommerceOrder(
 			commerceOrderLocalService.updatePaymentStatus(
 				userId, commerceOrder.getCommerceOrderId(), paymentStatus);
 
-			return commerceOrderLocalService.updateOrderStatus(
-				commerceOrder.getCommerceOrderId(), paymentStatus);
+			commerceOrder.setOrderStatus(paymentStatus);
```

Comment:
I know it was the same before, but this can't be right, we are settings the OrderStatus using the PaymentStatus. We should use payment for payment and order for order.